torch/_export/converter.py

```diff
@@ -937,22 +941,22 @@ def __init__(
         self.sample_args = sample_args
         self.sample_kwargs = sample_kwargs
 
-        self.name_to_param_map: Dict[str, torch.Tensor] = (
-            dict(ts_model.named_parameters())
-            if isinstance(ts_model, torch.jit.ScriptModule)
-            else {}
-        )
-        self.name_to_buffer_map: Dict[str, torch.Tensor] = (
-            dict(ts_model.named_buffers())
-            if isinstance(ts_model, torch.jit.ScriptModule)
-            else {}
-        )
+        self.name_to_param_map: Dict[str, torch.Tensor] = {}
+        self.name_to_buffer_map: Dict[str, torch.Tensor] = {}
+        if not isinstance(self.ts_model, torch._C.ScriptFunction):
+            for k, tensor in self.ts_model.state_dict().items():  # type: ignore[union-attr]
+                if tensor.requires_grad:
```

Comment:
I don't think this is the right way to check? Would seeing if the tensor is `nn.Parameter` work? Otherwise we can just throw everything in buffers?